tinylog-api-kotlin/src/main/kotlin/org/tinylog/kotlin/TaggedLogger.kt

```diff
@@ -56,9 +58,7 @@ class TaggedLogger internal constructor(private val tag: String?) {
 	 * Any object with a meaningful [Any.toString] method
 	 */
 	fun trace(message: Any?) {
-		if (minimumLevelCoversTrace) {
-			provider.log(stackTraceDepth, tag, Level.TRACE, null, null, message)
-		}
+		traceTags.forEach { provider.log(stackTraceDepth, it, Level.TRACE, null, null, message) }
```

Comment:
Does the the lambda call add stack trace element!? I'm not sure. Maybe we need to add `+1` to `stackTraceDepth`!?

Reply:
The call doesn't add a stack trace element, but I am thinking it might be better to refactor to use an explicit for-loop anyway.

The `forEach` call doesn't add a stack trace element because it is an [inline function](https://kotlinlang.org/docs/inline-functions.html#noinline) (see [Kotlin documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/for-each.html)). The compiler will inline a for-loop in. So, it shouldn't add a stack trace element in.

To further test this, I quickly tested this with the following code:
```kotlin
fun main(args: Array<String>) {
    doLog(1, 2, 3)
}

fun doLog(vararg values: Int) {
    values.toSet().forEach { v ->
        println(v)
        if (v == 3) {
            NullPointerException("Got 3").printStackTrace()
        }
    }
}
```

The output from this was:
```
1
2
3
java.lang.NullPointerException: Got 3
	at test.tinylog.LoopTesterKt.doLog(LoopTester.kt:13)
	at test.tinylog.LoopTesterKt.main(LoopTester.kt:6)
```

Notice the `forEach` didn't appear in the stack trace!

Anyway, this discussion was making me think about refactoring to using an explicit for-loop anyway because I think it will be easier to think about and it won't rely on `forEach` being an inline function.

Reply:
Thank you very much for the investigation and the great explanation. Great to know that it is an inline function :) I agree with you that we should not rely on it.